src/utils/segments-middleware.js

```diff
@@ -0,0 +1,37 @@
+import _ from "lodash";
+
+const fieldPath = "ship.private_settings.synchronized_segments";
```

Comment:
we can use `synchronized_segments` as a naming convention and default value here but let's make this configurable.